docs/book/src/architecture/controllers/machine.md

```diff
@@ -1 +1,67 @@
-# Machine
+# Machine  Controller
+
+The machine controller is responsible for keeping the machine object in sync with the infrastructure machine object as
+as well as managing references and properly deleting dependent objects.
+
+Its main responsibilities are:
+
+* Setting an OwnerReferences on various objects, including adopting machines into clusters.
+* Setting NodeRefs to be able to associate machines and kubernetes nodes.
+* Deleting all related objects in the correct order.
+* Keeping the Machine object up to date with external objects.
+
+## Contracts
+
+### Cluster API
+
+Cluster associations are made via labels.
+
+A machine must have a label with a key of `cluster.x-k8s.io/cluster-name` and a value of the cluster's name.
```

Comment:
Does it point to the`Cluster` object or the `MyProviderCluster` object?

Reply:
Cluster, thanks for asking!